docs/basics/101-130-yodaproject.rst

```diff
@@ -0,0 +1,564 @@
+.. index:: ! datalad command; create-sibling-gitlab
+
+The command takes a repository name and Github authentication credentials
+(either in the command line call with options ``github-login <NAME>`` and
+``github-passwd <PASSWORD>``, with an *oauth* token stored in the Git
+configuration [#f5]_, or interactively). Based on the credentials and the
+repository name, it will create a new, empty repository on Github, and
+configure this repository as a sibling of the dataset:
+
+.. runrecord:: _examples/DL-101-130-121
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+   :realcommand: datalad --log-level critical siblings add -d . --name github --url https://github.com/adswa/midtermproject.git
+
+   $ datalad create-sibling-github -d . midtermproject
+
+Verify that this worked by listing the siblings of the dataset:
+
+.. runrecord:: _examples/DL-101-130-122
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   $ datalad siblings
+
+On Github, you will see a new, empty repository with the name
+``midtermproject``. However, it does not yet contain any of your dataset's
+history or files. This requires *publishing* the current state of the dataset
+to this sibling:
+
+.. code-block:: bash
+
+   $ datalad publish --to github
+   [INFO] Publishing <Dataset path=/home/me/dl-101/DataLad-101/midterm_project> to github
+   publish(ok): . (dataset) [pushed to github: ['[new branch]', '[new branch]']]
+
+.. gitusernote::
+
+   Creating a sibling on Github will create a new empty repository under the
+   account that you provide and set up a *remote* to this repository. Upon a
+   :command:`datalad publish` to this sibling, your datasets history
+   will be pushed there.
+
+Yay! Consider your midterm project submitted! Others can now install your
+dataset and check out your data science project -- and even better: they can
+reproduce your data science project easily from scratch!
+
+.. findoutmore:: On the looks and feels of this published dataset
+
+   Now that you have created and published such a YODA-compliant dataset, you
+   are understandably excited how this dataset must look and feel for others.
+   Therefore, you decide to install this dataset in a new location on your
+   computer, just to get a feel for it.
+
+   Replace the ``url`` in the :command:`install` command below with the path
+   to your own ``midtermproject`` Github repository:
+
+   .. runrecord:: _examples/DL-101-130-125
+      :language: console
+      :workdir: dl-101/DataLad-101/midterm_project
+
+      $ cd ../../
+      $ datalad install -r --source "https://github.com/adswa/midtermproject.git"
+
+   Note that we performed a *recursive* installation by providing the ``-r``
+   option. Thus, we don't need to install the ``input/`` subdataset anymore.
+   Let's start with the subdataset, and see whether we can retrieve the
+   input ``iris.csv`` file. This should not be a problem, since it's origin
+   is recorded:
+
```

Comment:
Unclear why this works. I think it might be only because we created the iris dataset manually above. It might be more useful if we put that dataset on github directly, and install it from there to begin with?

Reply:
FTR I published the dataset to Github once and then replace the interactive code snippet that was executed with a static code-block. But the dataset currently lives in ``github.com/adswa/midtermproject``.